base/game_events/DestroyerEvent.ts

```diff
@@ -0,0 +1,33 @@
+import {GameEvent, event_types} from "./GameEvent";
+
+export class DestroyerEvent extends GameEvent {
+    private target_event_key: string;
+
+    //these were the basic things that every child class had, so I took it.
+    //though the destroyer is just like a trigger, press it and it kills an event.
+    //it only needs target_event_key imo.
+    constructor(game, data, active, key_name) {
+        super(game, data, event_types.DESTROYER, active, key_name);
+    }
+
+    set_target_event(target_event_key: string) {
```

Comment:
Instead of adding a method to set the target event label, pass `target_event_key` in the constructor and assign it to `DestroyerEvent.target_event_key` there after `super`.